packages/fuels-core/src/encoding_utils.rs

```diff
@@ -0,0 +1,110 @@
+use crate::{EnumVariants, ParamType, WORD_SIZE};
+
+// Calculates how many WORDs are needed to encode any given variant in `variants`
+pub fn encoding_width_to_fit_any(variants: &EnumVariants) -> usize {
```

Comment:
`compute_max_needed_encoding_width` would make more sense imo.

Reply:
I went in another direction, this is now called:
`compute_encoding_width_of_enum`

and will include the discriminant size.